appinventor/components/src/com/google/appinventor/components/runtime/Notifier.java

```diff
@@ -332,6 +349,7 @@ public void onClick(DialogInterface dialog, int which) {
           new DialogInterface.OnClickListener() {
             public void onClick(DialogInterface dialog, int which) {
               HideKeyboard((View) input);
+              TextInputCanceled();
```

Comment:
@ColinTree @ewpatton Perhaps TextInputCanceled() should return the value of EventDispatcher.dispatchEvent. If it is true, that means that the project had a TextInputCanceled handler which handled the event and in that case perhaps we should not call AfterTextInput(cancelButtonText).

Reply:
@jisqyv One potential problem that might occur with the proposed suggestion is the following:

1. Build an app using the old `AfterTextInput` event
2. Upgrade project
3. Pull in new `TextInputCanceled` event
4. Move the new event off to the side (e.g., I have some people attempt to drag the event back into the palette to "delete" it, so it continues to exist in the project off screen).

Because `TextInputCanceled` exists, the `AfterTextInput` event will no longer be called even though there are blocks to handle the event. The reason why the semantics change may be non-obvious. The user will need to find and either delete the `TextInputCanceled` event or migrate the logic from the `AfterTextInput` event to the `TextInputCanceled` event.

We could consider deprecating the existing event and provide a warning that checks for the presence of both event blocks going forward if we decide this is a route we want to take.

Reply:
@ewpatton What are you proposing to deprecate? AfterTextInput is also called when there is actual text input :-)

Reply:
@jisqyv Ah, right. We should not deprecate. I still do not want to force everyone to update the semantics of their existing apps by removing the call to `AfterTextInput` here.

Reply:
@ewpatton They shouldn't have to update their semantics. If they do not drag out a TextInputCanceled block, things will work as before. I understand your concern that someone might drag one out and not understand the implication (that AfterTextInput will now not get called). Frankly I don't feel that strongly about this, so if we want to leave the call as is, then fine.